CMakeLists.txt

```diff
@@ -0,0 +1,134 @@
+# Copyright 2023 Viam Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# TODO: [required] libviam_rust_utils handling
```

Comment:
I didn't remove this `TODO` for `libviam_rust_utils` because it isn't quite complete:
- As noted where I do the setup for it, I'd like to push the logic I wrote to either use local or download down into a tool so it doesn't clutter the top-level `CMakeLists.txt`
- The current dylib you get from the GH releases page has some RPATH issues. Those will hopefully get sorted out when I tackle the other TODO here to make `BUILD_SHARED_LIBS` work.

Also, as a more general question, do we have plans afoot to push the rust_utils library up to homebrew or similar? If we do, we should try to write a `.cmake` file for it so it can be found via `find_package`.

Reply:
Yes, agreed. As soon as this project can consume `rust_utils` as an external package I'd immediately remove the download step from this project. But until that happens I want to be able to deliver a "batteries included" experience to C++ SDK users.

I have a few tickets I'd like to file for `rust_utils`. The issues/improvements I'd like to see are:
- Adjust the release naming on the downloads page from `macosx` to `Darwin` as noted elsewhere in this review.
- Provide a minimal C header describing the functions exported from `rust_utils`, so that packages like this one do not need to constantly reiterate declaring `init_rust_runtime`, etc.
- Set a SONAME/installname and install associated versioned symlinks.
- Provide `.pc` and `.cmake` files for `rust_utils` so it can be found and used via both `pkg-config` and CMake's `find_package` infra.
- Publish it to `homebrew`, and hopefully later see it upstreamed into Debian.

What is the right JIRA project in which to file tickets for these items?

Reply:
RSDK is the right project for most bugs and small TODOs. Major improvements and projects typically need to go through Product first though. Note that there's an "SDK/Netcode" team, AND a "RustUtils" component and so you probably want to set both of those on relevant tickets.

Reply:
I will be filing several RSDK tickets about `viam_rust_utils`.